### PR TITLE
DEV-823: update error msg for project creation in SIO

### PR DESCRIPTION
**Pull request by tranlyvu**

### CODE REVIEW

**Comment by ChrisZhong** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [AdapterDeterLab.java#L637](https://github.com/nus-ncl/services-in-one/blob/8dea519d72c1c8ca23f6f7315fc6414c5e4549c7/adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java#L637): The Cyclomatic Complexity of this method "checkAdapterResultNewUsers" is 15 which is greater than 10 authorized. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by ChrisZhong** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [AdapterDeterLab.java#L681](https://github.com/nus-ncl/services-in-one/blob/54c37e3c5f6a04ec39dccc0b76090805a202f764/adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java#L681): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1067)

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.springframework.web.bind.annotation.ResponseStatus;
+import sg.ncl.common.exception.base.BadRequestException;
+
+/**
+ * Created by Tran Ly Vu on 11/21/2016.
+ */
+
+@ResponseStatus(reason = "Verification key error")
+public class VerificationKeyException extends BadRequestException {
```
Comment (ZHANGChunwang) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java]: This is not a BadRequestException. It's something error on the Deter side. So could reuse AdapterDeterLabOperationFailedException

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/TeamNameExistsException.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import sg.ncl.common.exception.base.BadRequestException;
+/**
+ * Created by Tran Ly Vu on 11/21/2016.
+ */
+
+@ResponseStatus (value = HttpStatus.BAD_REQUEST, reason = "Team Name is already in use")
+public class TeamNameExistsException  extends BadRequestException{
```
Comment (ZHANGChunwang) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/TeamNameExistsException.java]: There is an exception called TeamNameAlreadyExistsException. Please use that one.

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java [hunk 3]
```diff
@@ -635,17 +635,60 @@ public String getUsageStatistics(String teamId, String startDate, String endDate
...
+        }
+    }
+
+    private void checkEmailAddressError (String responseBody, String jsonResult, String logPrefix) {
+
         if ("email address in use".equals(jsonResult)) {
-            log.warn(logPrefix + " Email address already exists.", responseBody);
+            log.warn(logPrefix + "Email address already exists.", responseBody);
             throw new EmailAlreadyExistsException();
         } else if ("invalid password".equals(jsonResult)) {
```
Comment (ZHANGChunwang) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: "invalid password" should not be checked here. Please use a new method called checkInvalidPasswordError(...)

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java [hunk 4]
```diff
@@ -635,17 +635,60 @@ public String getUsageStatistics(String teamId, String startDate, String endDate
...
-            log.warn(logPrefix + " Password is invalid.", responseBody);
+            log.warn(logPrefix + "Password is invalid.", responseBody);
             throw new InvalidPasswordException();
-        } else if (!"user is created".equals(jsonResult)) {
-            log.warn(logPrefix, responseBody);
-            throw new DeterLabOperationFailedException();
+        }
+    }
+
+    private void checkTeamNameError (String responseBody, String  jsonResult, String logPrefix) {
```
Comment (ZHANGChunwang) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: Please rename method to checkTeamNameAlreadyExistsError(...)

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java [hunk 5]
```diff
@@ -635,17 +635,60 @@ public String getUsageStatistics(String teamId, String startDate, String endDate
...
+
+        if ("team name is already in use".equals(jsonResult)) {
+            log.warn(logPrefix + "Team Name is already in use.", responseBody);
+            throw new TeamNameExistsException();
+        }
+    }
+
+    private void checkUserNotFound (String responseBody, String jsonResult, String logPrefix) {
```
Comment (ZHANGChunwang) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: Please rename method to checkUserNotFoundError(...)

Comment (tranlyvu) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: fixed

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java (hunk 4 above)
Comment (tranlyvu) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: fixed

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java (hunk 1 above)
Comment (tranlyvu) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java]: fixed

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/TeamNameExistsException.java (hunk 2 above)
Comment (tranlyvu) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/TeamNameExistsException.java]: did not fix, you can just approve first, then we fix later after merge

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package sg.ncl.adapter.deterlab.exceptions;
+
+import org.springframework.web.bind.annotation.ResponseStatus;
+import sg.ncl.common.exception.base.BadRequestException;
```
Comment (ChrisZhong) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java]: MINOR Remove this unused import 'sg.ncl.common.exception.base.BadRequestException'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Created by Tran Ly Vu on 11/21/2016.
+ */
+
+@ResponseStatus(reason = "Verification key error")
+public class VerificationKeyException extends DeterLabOperationFailedException {
```
Comment (ChrisZhong) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/VerificationKeyException.java]: MAJOR This class has 6 parents which is greater than 5 authorized. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)

**Comment by ChrisZhong** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java [hunk 8]
```diff
@@ -69,10 +75,13 @@ public String joinProjectNewUsers(String jsonString) {
         // msg: user not found, uid: xxx
         String responseBody = response.getBody().toString();
         try {
-            checkAdapterResultNewUsers(responseBody, "Join new project as new user failed: {}.");
-
-            log.info("Join project as new user to DeterLab OK");
-            return responseBody;
+            //checkAdapterResultNewUsers(responseBody, "Join new project as new user failed: {}.");
```
Comment (ChrisZhong) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by ChrisZhong** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Pull request merged**